mex/backend/auxiliary/wikidata.py

```diff
@@ -0,0 +1,70 @@
+from functools import cache
+from typing import Annotated
+
+from fastapi import APIRouter, Query
+
+from mex.backend.auxiliary.models import PagedResponseSchema
+from mex.common.models import ExtractedOrganization, ExtractedPrimarySource
+from mex.common.primary_source.extract import extract_seed_primary_sources
+from mex.common.primary_source.transform import (
+    get_primary_sources_by_name,
+    transform_seed_primary_sources_to_extracted_primary_sources,
+)
+from mex.common.types import TextLanguage
+from mex.common.wikidata.extract import search_organizations_by_label
+from mex.common.wikidata.transform import (
+    transform_wikidata_organizations_to_extracted_organizations,
+)
+
+router = APIRouter()
+
+
+@router.get("/wikidata", status_code=200, tags=["wikidata"])
+def search_organization_in_wikidata(
+    q: Annotated[str, Query(min_length=1, max_length=1000)],
+    offset: Annotated[int, Query(ge=0, le=10e10)] = 0,
+    limit: Annotated[int, Query(ge=1, le=100)] = 10,
+    lang: TextLanguage = TextLanguage.EN,
+) -> PagedResponseSchema[ExtractedOrganization]:
+    """Search an organization in wikidata.
+
+    Args:
+        q: label of the organization to be searched
+        offset: start page number
+        limit: end page number
+        lang: language of the label. Example: en, de
+
+    Returns:
+        Paginated list of ExtractedOrganization
+    """
+    organizations = search_organizations_by_label(q, offset, limit, lang)
+
+    extracted_organizations = list(
+        transform_wikidata_organizations_to_extracted_organizations(
+            organizations, extracted_primary_source_wikidata()
+        )
+    )
+
+    return PagedResponseSchema(
+        total=len(extracted_organizations),
+        offset=offset,
+        limit=limit,
```

Comment:
`total` is supposed to be the total number of search hits found on wikidata, not the number of returned items, any request-sender could figure that out by itself. (same is true for offset and limit, we don't need to mirror that back.)

e.g. if you'd search for "institute" you might get a couple of hundred hits on wikidata, but we only return `limit` / up to 100 of them. and a UI would need to know how many there are overall, to display a proper pagination bar.

i know this might complicate this PR a lot and i'm sorry i didn't spot that earlier in the mex-common pr. so at this point, i'd suggest you add a stop-gap comment in this line and create a new ticket stub to return the correct pagination total later

Reply:
updated in 561e341b97850ed524e6346cdc1cfcf6b22f1a8b